valhalla/baldr/double_bucket_queue.h

```diff
@@ -60,8 +61,8 @@ class DoubleBucketQueue {
    * @param   label  Label index to add to the queue.
    * @param   cost   Cost for this label.
    */
-  void add(const uint32_t label, const float cost) {
-    get_bucket(cost).push_back(label);
+  void add(const uint32_t label) {
+    get_bucket(labelcost_(label)).push_back(label);
```

Comment:
this is the important part. it means that like `decrease` add gets the labels cost from the label itself, so they can never disagree on what that cost is